main.go

```diff
@@ -1,30 +1,261 @@
 package main
 
 import (
+	"context"
+	"encoding/base64"
 	"fmt"
+	"io"
 	"os"
 
 	"github.com/alecthomas/kong"
+	"github.com/pkg/errors"
+	"google.golang.org/grpc"
+	rpb "google.golang.org/grpc/reflection/grpc_reflection_v1alpha"
+	"google.golang.org/protobuf/encoding/protojson"
+	"google.golang.org/protobuf/encoding/prototext"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/reflect/protoreflect"
 )
 
 var version = "v0.0.0"
 
+type globals struct {
+	Address   string `short:"a" env:"GRPC_ADDRESS" help:"gRPC server address"`
+	Plaintext bool   `short:"p" help:"Use plain-text; no TLS" env:"GRPC_PLAINTEXT"`
+	Format    string `short:"f" help:"output protoset as one of json, base64, bin, text" enum:"json,base64,bin,text" default:"json"`
+	Out       string `short:"o" help:"output file, default: stdout" default:"-"`
+
+	out         io.Writer
+	hostAddress string // used in tests to work with localhost:0
+}
+
 type config struct {
-	Address   string           `short:"a" env:"GRPC_ADDRESS" help:"gRPC server address"`
-	Plaintext bool             `short:"p" help:"Use plain-text; no TLS" env:"GRPC_PLAINTEXT"`
-	Version   kong.VersionFlag `short:"V" help:"Print version information" group:"Other:"`
+	globals
+	Version    kong.VersionFlag `short:"V" help:"Print version information" group:"Other:"`
+	Services   servicesCmd      `cmd:"" help:"Call list_services"`
+	Symbol     symbolCmd        `cmd:"" help:"Call file_containing_symbol"`
+	Filename   filenameCmd      `cmd:"" help:"Call file_by_filename"`
+	Extension  extensionCmd     `cmd:"" help:"Call file_containing_extension"`
+	Extensions extensionsCmd    `cmd:"" help:"Call all_extension_numbers_of_type"`
+}
+
+type (
+	servicesCmd struct{}
```

Comment:
a little bit odd that you've grouped these two types but no others.

Reply:
yes, oups, oversight, fixed.